src/composables/forms.ts

```diff
@@ -252,6 +252,20 @@ export const emailPattern = String.raw`^\w+([\-+.']\w+)*@\w+([\-.]\w+)*\.\w+([\-
  */
 export const phonePattern = String.raw`[0-9]{10}|[0-9]{3}-[0-9]{3}-[0-9]{4}`
 
+/**
+ * url validation regex pattern
+ * used with input type="url" in the pattern attribute for html5
+ *
+ * The pattern is conservative and focuses on ensuring the protocol is valid
+ * and the remainder of the url does not include empty spaces.  If additional protocols
+ * need to be supported, they can be added to the first group (?:https|http|ftp).
+ *
+ * Reference: https://regex101.com/r/P5dlas/3
+ *
+ * NOTE(spk): not all JavaScript regular expression characters are not supported in input patterns.
+ */
+export const urlPattern = String.raw`^(?:https|http):\/\/[^ ]+$`
```

Comment:
I kept this on the conservative side, since a lack of a protocol or protocol mangling are the most common issues.  This does tightly scope our `<BaseInput type="url" />` to Website URL's, but that appears to be our only use case, and I can't remember the last time I needed to support another protocol.  

However, you can always reach for type="text" with a custom pattern. ex:  `<BaseInput type="text" pattern="^(?:sftp|ftp):\/\/[^ ]+$" />`